torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -659,36 +664,60 @@ def _profiler_test_with_rpc(self, rpc_exec_mode, func, args):
             self.assertTrue(func.__name__ in rpc_event.name)
             self.assertTrue(rpc_exec_mode.value in rpc_event.name)
             self.assertEqual(rpc_event.count, 1)
+            if use_record_function:
+                # verify order by ensuring that the outer context comes
+                # before the rpc event.
+                foo_event_ix = next(i for i, event in enumerate(events) if "foo" in event.name)
```

Comment:
Curious, besides the order, is there anything else we can get from the profiling result of "foo"?

Reply:
We should be guaranteed that the CPU time of foo_event_ix is greater than the RPC since it should be a superset. Let me look into adding this into the test.